core/controllers/reader.py

```diff
@@ -213,6 +213,25 @@ def get(self, exploration_id):
 
         exploration_rights = rights_manager.get_exploration_rights(
             exploration_id, strict=False)
+
+        # Retrieve all classifiers for the exploration.
+        state_classifier_mapping = {}
+        classifier_training_jobs = (
+            classifier_services.get_classifier_training_jobs(
+                exploration_id, exploration.version, exploration.states))
+        for index, state_name in enumerate(exploration.states):
+            if classifier_training_jobs[index] == None:
+                state_classifier_mapping[state_name] = None
+            else:
+                # Not making this a multi-get since we are deprecating
```

Comment:
Ah, I see -- thanks for explaining!

OK, in that case, by the time this PR is merged, classifier_data does need to sit in the job model so that we only do a single get_multi() to get all classifier jobs, and nothing else. It is important that the learner view loads fast.